cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -1039,6 +1039,7 @@ def _add_head_node(self):
                     "enable_intel_hpc_platform": "true" if self.config.is_intel_hpc_platform_enabled else "false",
                     "cw_logging_enabled": "true" if self.config.is_cw_logging_enabled else "false",
                     "log_rotation_enabled": "true" if self.config.is_log_rotation_enabled else "false",
+                    "detailed_monitoring_enabled": "true" if self.config.is_detailed_monitoring_enabled else "false",
```

Comment:
I think we don't need to pass it into dna.json, see reason in the comment below

Reply:
If I remove this, `test_cluster_stack.py:test_head_node_dna_json` would fail.
I can remove the other ones since no unit tests are testing them (assuming that we don't expect this new parameter in userdata, rather than that we're missing a test).